sumatra/core.py

```diff
@@ -87,16 +87,36 @@ def _get_process_children(pid):
     return [int(child_pid) for child_pid in stdout.split()]
 
 
-class Registry(object):
+class _SingletonType(type):
```

Comment:
A leading underscore implies `_SingletonType` should not be used outside the current module. However, it is used in at least one of the test scripts, so I would argue for naming it `SingletonType`.